src/model/implementation/model.ts

```diff
@@ -42,23 +43,26 @@ export class Model implements ModelComponent {
             allPermissionProfiles: this.permissionProfiles
         });
         this.namespaces = [this.rootNamespace, ...this.rootNamespace.descendantNamespaces];
-        this.typeMap = new Map(this.types.map((type): [string, Type] => ([type.name, type])));
+        this.typeMap = new Map(this.types.map((type): [string, Type] => [type.name, type]));
         this.i18n = new ModelI18n(input.i18n || [], this);
+        this.billingEntityTypes = input.billing
+            ? input.billing.billingEntities.map(value => new BillingEntityType(value, this))
+            : [];
+        this.modelValidationOptions = input.modelValidationOptions;
     }
 
     validate(context = new ValidationContext()): ValidationResult {
         this.validateDuplicateTypes(context);
 
         this.i18n.validate(context);
 
+        this.billingEntityTypes.forEach(value => value.validate(context));
```

Comment:
I prefer `for-of` looks over `forEach`, it makes it clear that there are side-effects, it has better support for type narrowing in typescript, and it avoids missing / adding broken `await`s.